sormas-app/app/src/main/java/de/symeda/sormas/app/report/aggregate/AggregateReportsFragment.java

```diff
@@ -351,6 +365,24 @@ private void showReportData() {
 		}
 	}
 
+	private boolean isDuplicateByDiseaseEpiWeekAndInfrastructure(
+		Disease disease,
+		EpiWeek epiWeek,
+		InfrastructureAdo selectedInfrastructure,
+		User selectedUser) {
+
+		List<AggregateReport> duplicates = reports.stream()
+			.filter(
```

Comment:
I'd rather use `anyMatch` here, hoping that it would perform slightly better because the whole process could then abort as soon as a duplicate is found

Reply:
that was my initial approach but during testing I realized that this approach does not cover all the situations. In the return statement, I test the duplicates size and actually in the logic there is a difference between one result and more than one results so anyMatch will not work here.